### PR TITLE
Replace `DateTime` with `DateTimeImmutable`

### LINKED ISSUE #21: DataTime imutable
One more suggestion - replace `DateTime` with [`DateTimeImmutable`](https://www.php.net/manual/en/class.datetimeimmutable.php) in all code places. Reason - date/time seems like immutable during API calls

### PR DESCRIPTION
**Pull request by gomzyakov**
### Changed

- Replace `DateTime` with `DateTimeImmutable` in all code places

Fixes #21 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: tests/Unit/Responses/Entities/EntitiesFactory.php [hunk 1]
```diff
@@ -82,19 +83,19 @@ protected static function bootUpFactories(): void
...
                 'active_from' => $as_array
-                    ? DateTimeFactory::toIso8601Zulu($faker->dateTimeThisMonth)
-                    : $faker->dateTimeThisMonth,
+                    ? DateTimeFactory::toIso8601Zulu(DateTimeImmutable::createFromMutable($faker->dateTimeThisMonth))
+                    : DateTimeImmutable::createFromMutable($faker->dateTimeThisMonth),
                 'active_to'   => $as_array
-                    ? DateTimeFactory::toIso8601Zulu($faker->dateTimeThisMonth)
-                    : $faker->dateTimeThisMonth,
+                    ? DateTimeFactory::toIso8601Zulu(DateTimeImmutable::createFromMutable($faker->dateTimeThisMonth))
+                    : DateTimeImmutable::createFromMutable($faker->dateTimeThisMonth),
```
Comment (gomzyakov) [inline tests/Unit/Responses/Entities/EntitiesFactory.php:98]: I can get refactoring of this code & similar code below.

But, this is beyond the scope of the task, right?

File: tests/Unit/ClientTest.php [hunk 2]
```diff
@@ -157,7 +157,7 @@ public function testDoRequestWithUnknownReportTypeUid(): void
             new Response(
                 500, ['content-type' => 'application/json;charset=utf-8'], \json_encode((object) [
                     'state' => $state = 'fail',
-                    'stamp' => $stamp = DateTimeFactory::toIso8601Zulu(new DateTime),
+                    'stamp' => $stamp = DateTimeFactory::toIso8601Zulu(new DateTimeImmutable()),
```
Comment (eldario) [inline tests/Unit/ClientTest.php]: del `()`

File: tests/Unit/ClientTest.php [hunk 3]
```diff
@@ -404,7 +404,7 @@ public function testDevToken(): void
                 'is_hash' => ($is_hash = $this->faker->boolean)
                     ? 'true'
                     : 'false',
-                'date'    => DateTimeFactory::toIso8601ZuluWithoutMs($date = new DateTime),
+                'date'    => DateTimeFactory::toIso8601ZuluWithoutMs($date = new DateTimeImmutable()),
```
Comment (eldario) [inline tests/Unit/ClientTest.php]: del `()`

**Review by eldario — CHANGES_REQUESTED** [review]

File: tests/Unit/ClientTest.php (hunk 3 above)
Comment (gomzyakov) [reply tests/Unit/ClientTest.php]: Fixed 6b3bd06c0b6414c6ee11bc614c2003c1b22a17ee

File: tests/Unit/ClientTest.php (hunk 2 above)
Comment (gomzyakov) [reply tests/Unit/ClientTest.php]: Fixed 6b3bd06c0b6414c6ee11bc614c2003c1b22a17ee

**Comment by tarampampam** [comment]
Blocked until #22 is not merged

**Pull request closed without merging**